### PR TITLE
POM-43 [B] Zaprojektować kontrakt API

### PR DESCRIPTION
**Pull request by MaaKaa**
* Add springdoc library.
* Implement MaterialAidAdController & MaterialAidAdDto + validation.

### CODE REVIEW

**Comment by sfc-gh-pnescior** [comment]
Hi, I've already added springdoc in https://github.com/coi-gov-pl/pomocua-ogloszenia/pull/8
I configured it in a way, that only enabling `api-docs` spring-boot profile will enable those docs, to be sure that they are not enabled on the production by mistake.

File: src/main/java/pl/gov/coi/pomocua/ads/materialaid/domain/dto/MaterialAidOfferDto.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.io.Serializable;
+
+@Setter
+@Getter
+@Builder
+public class MaterialAidOfferDto implements Serializable {
+
+    @Serial
+    private static final long serialVersionUID = 4899363368417701147L;
+    private static final String EXCLUDING_SPECIAL_CHARS = "^[^<>()%#@”‘]*$";
```
Comment (sfc-gh-krzymkowski) [inline src/main/java/pl/gov/coi/pomocua/ads/materialaid/domain/dto/MaterialAidOfferDto.java:20]: I know these chars where specified in the docs, but I don't see why we should be worried about those chars:
* % - there is no sql injection risk, as we're not building queries by contatenation
* <> - Angular will escape those, so XSS risk in handled

This rule is way too strict. These are chars that should be allowed in the description/content.

File: src/main/java/pl/gov/coi/pomocua/ads/materialaid/domain/dto/MaterialAidOfferDto.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+import java.io.Serial;
+import java.io.Serializable;
+
+@Setter
+@Getter
+@Builder
+public class MaterialAidOfferDto implements Serializable {
```
Comment (sfc-gh-krzymkowski) [inline src/main/java/pl/gov/coi/pomocua/ads/materialaid/domain/dto/MaterialAidOfferDto.java:16]: We've introduced a `BaseOffer` entity that has id, title and description. I recommend you extend from it.

File: src/main/java/pl/gov/coi/pomocua/ads/materialaid/api/MaterialAidAdController.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/api/secure/material-aid")
+public class MaterialAidAdController {
+
+    @Operation(description = "Saves material aid offer")
+    @PostMapping(produces = "application/json", consumes = "application/json")
+    public ResponseEntity<Object> postMaterialAidOffer(@Valid @RequestBody final MaterialAidOfferDto materialAidOfferDto) {
```
Comment (sfc-gh-krzymkowski) [inline src/main/java/pl/gov/coi/pomocua/ads/materialaid/api/MaterialAidAdController.java:20]: See for example JobsResource for a complete controller implementation (with tests)

File: src/main/java/pl/gov/coi/pomocua/ads/materialaid/domain/dto/MaterialAidOfferDto.java (hunk 1 above)
Comment (MaaKaa) [reply src/main/java/pl/gov/coi/pomocua/ads/materialaid/domain/dto/MaterialAidOfferDto.java:20]: Discuss it with the business side then ;)

File: src/main/java/pl/gov/coi/pomocua/ads/materialaid/domain/dto/MaterialAidOfferDto.java (hunk 2 above)
Comment (MaaKaa) [reply src/main/java/pl/gov/coi/pomocua/ads/materialaid/domain/dto/MaterialAidOfferDto.java:16]: Ok. It hasn't been implemented yet when I pulled the repo.

**Pull request closed without merging**